core/src/main/java/nl/nn/adapterframework/jdbc/dbms/SqlTranslator.java

```diff
@@ -0,0 +1,162 @@
+	private final Logger log = LogUtil.getLogger(this);
+
+	private static final String PATTERN_FILE = "sql-translate.properties";
+
+	private Map<String,Pattern> sources;
+	private Map<String,String>  targets;
+	
+	private boolean configured=false;
+
+	public SqlTranslator(String source, String target) throws JdbcException {
+		if (StringUtils.isEmpty(source) || StringUtils.isEmpty(target))
+			throw new IllegalArgumentException("Can not translate from [" + source + "] to [" + target + "]");
+		if (source.equalsIgnoreCase(target)) {
+			log.warn("Same source and target for SqlTranslator. Skipping pattern generation.");
+			return;
+		}
+		try {
+			if (!readPatterns(source,target)) {
+				return;
+			}
+		} catch (Exception e) {
+			throw new JdbcException("cannot create SqlTranslator",e);
+		}
+		configured = true;
+	}
+
+	@Override
+	public boolean canConvert(String from, String to) {
+		return configured;
+	}
+	
+	/**
+	 * Translates the given query to the target language.
+	 * Uses the translation rules set by this and the target translators.
+	 *
+	 * @param original Original query to be translated.
+	 * @return Translated query.
+	 */
+	@Override
+	public String translate(String original) {
+		String query = original;
+		if (sources!=null) {
+			for (String label:sources.keySet()) {
+				Matcher matcher = sources.get(label).matcher(query);
+				if (matcher.find()) {
+					if (log.isTraceEnabled()) log.trace(String.format("Found a match for label [%s] pattern [%s]",label, sources.get(label)));
+					String replacement = targets.get(label);
+					if (StringUtils.isNotEmpty(replacement)) {
+						query = matcher.replaceAll(replacement);
+					} else {
+						query = matcher.replaceAll("");
+					}
+				}
+			}
+		}
+		if (StringUtils.isEmpty(query)) {
+			return null;
+		}
+		return query;
+	}
+
+	/**
+	 * Compiles a pattern with necessary flags.
+	 * @param str String to be compiled.
+	 * @return Output pattern.
+	 */
+	protected Pattern toPattern(String str) {
+		// Make sure there are no greedy matchers.
+		String pattern = str.replaceAll("\\.\\*", ".*?");
+		log.trace("Compiling pattern [" + pattern + "]");
+		return Pattern.compile(pattern, Pattern.CASE_INSENSITIVE | Pattern.MULTILINE);
+	}
+
+	/**
+	 * Reads data from SOURCE_CSV file.
+	 * Puts the data in memory to be used later.
+	 * @throws IOException If database name can not be found or file can not be read.
+	 */
+	private boolean readPatterns(String sourceDialect, String targetDialect) throws IOException {
+		sources = new LinkedHashMap<>();
+		targets = new LinkedHashMap<>();
+
+		String sourceMatch=".source."+sourceDialect.replaceAll(" ", "_");
+		String targetMatch=".target."+targetDialect.replaceAll(" ", "_");
+
+		URL resourceUrl = ClassUtils.getResourceURL(Thread.currentThread().getContextClassLoader(), PATTERN_FILE);
+
+		try (BufferedReader reader = new BufferedReader(ClassUtils.urlToReader(resourceUrl, 10000))) {
+			String line= reader.readLine();
```

Comment:
Ik vind het raar om een timeout mee te geven voor het lokaal ophalen van een bestand.  Zie ook dat de betreffende methode nog nergens anders gebruikt wordt. De methode an zich is wel handig maar dan wel de connectietimeout argumenten verwijderen. 

In de methode zit een catch om UnsupportedEncodingException heen en vervolgens wordt een IOException gegooid, is ook een beetje raar toch? Die inheriten elkaar...